products/firestore/api.yaml

```diff
@@ -108,3 +108,80 @@ objects:
+    description: |
+      In Cloud Firestore, the unit of storage is the document. A document is a lightweight record
+      that contains fields, which map to values. Each document is identified by a name.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/firestore/docs/manage-data/add-data'
+      api: 'https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.documents'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: 'database'
+        default_value: '(default)'
+        description: |
+          The Firestore database id. Defaults to `"(default)"`.
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: 'collection'
+        description: |
+          The collection ID, relative to database. For example: chatrooms or chatrooms/my-document/private-messages.
+        required: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: 'documentId'
+        description: |
+          The client-assigned document ID to use for this document durring creation.
+        required: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        output: true
+        description: |
+          A server defined name for this index. Format:
+          `projects/{{project_id}}/databases/{{database_id}}/documents/{{path}}/{{document_id}}`
+      - !ruby/object:Api::Type::String
+        name: path
+        output: true
+        description: |
+          A relative path to the collection this document exists within
+      - !ruby/object:Api::Type::String
+        # This is a string instead of a NestedObject because fields can be deeply nested
+        name: fields
+        required: true
+        description: |
+          The document's [fields](https://firebase.google.com/docs/firestore/reference/rest/v1/projects.databases.documents) formated as a json string.
+
+          The map keys represent field names.
+
+          A simple field name contains only characters a to z, A to Z, 0 to 9,
+          or _, and must not start with 0 to 9. For example, foo_bar_17.
+
+          Field names matching the regular expression __.*__ are reserved.
+          Reserved field names are forbidden except in certain documented contexts.
+          The map keys, represented as UTF-8, must not exceed 1,500 bytes and cannot be empty.
+
+          Field paths may be used in other contexts to refer to structured fields defined here.
+          For mapValue, the field path is represented by the simple or quoted field names of the
+          containing fields, delimited by .. For example, the structured field "foo" : { mapValue: { "x&y" : { stringValue: "hello" }}}
+          would be represented by the field path foo.x&y.
+
+          Within a field path, a quoted field name starts and ends with ` and may contain any character.
```

Comment:
debating just linking to the documentation here... thoughts @rileykarson

Reply:
Yeah, I'd consider just linking the docs here. What's interesting is that this is a JSON string, and that gets obscured by all these rules.